python/cugraph-nx/cugraph_nx/algorithms/components/strongly_connected.py

```diff
@@ -0,0 +1,36 @@
+# Copyright (c) 2023, NVIDIA CORPORATION.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import cupy as cp
+import networkx as nx
+import numpy as np
+import pylibcugraph as plc
+
+from cugraph_nx.convert import to_directed_graph
+from cugraph_nx.utils.decorators import not_implemented_for
+
+__all__ = ["is_strongly_connected"]
+
+
+@not_implemented_for("undirected")
+def is_strongly_connected(G):
+    G = to_directed_graph(G)
+    N = len(G)
+    if N == 0:
```

Comment:
why not check graph size on the Nx graph before converting it to GPU?

Reply:
Being more intellegent when to convert graphs is a good idea, but this logic likely belongs elsewhere (TBD). On the other hand, small graphs should be quick regardless.